### PR TITLE
Add Litellm support

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Description

Add support for Litellm.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## How Has This Been Tested?

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ posthog = "^3.5.0"
 groq = "^0.9.0"
 together = "^1.2.1"
 boto3 = "^1.34.144"
+litellm = "^1.41.24"
```
Comment (deshraj) [inline pyproject.toml:26]: Lets make all of these as non-mandatory dependency on mem0ai similar to how we did for Embedchain.

Comment (deshraj) [reply pyproject.toml:26]: Actually feel free to do that as part of a follow up PR. Happy to merge this one.

**Review by deshraj — APPROVED** [review]
Looks good. Can you please bump up the version for Mem0 package?

File: pyproject.toml (hunk 1 above)
Comment (Dev-Khant) [reply pyproject.toml:26]: Yes will make those changes in next PR

**Review by deshraj — APPROVED** [review]

**Pull request merged**